### DISCUSSION TITLE
Coverage on multi-threaded applications

### DISCUSSION THREAD

**Discussion by stbla** [CodeIntelligenceTesting/jazzer#411] [Q&A]
Hi, I saw in the README that while Jazzer technically can fuzz multi-threaded targets, the "coverage can be misleading". I tried using Jazzer on a multi-threaded target and am getting 0 coverage in any code touched by threads other than the main thread. Is that intended behavior? I assumed that since all the threads are supposed to be recording coverage to some shared global variable, they would still be recording _some_ amount of coverage.

Thanks!

**Comment by fmeum** [comment]
We currently don't have any tests that cover this scenario, but at least in theory Jazzer should perform exactly as you describe: The coverage information would be subject to data races more or less by design, but those should be rather benign (loosing a +1 here and there) rather than essential (never reaching a non-zero count).

Do you have a small or at least public example that reproduces this behavior? I would be interested in taking a look.
